modules/oe_translation_poetry/tests/Kernel/ContactFieldsTest.php

```diff
@@ -0,0 +1,191 @@
+    parent::setUp();
+
+    $this->installSchema('node', ['node_access']);
+
+    $this->container->get('entity_type.manager')->getStorage('node_type')->create([
+      'type' => 'page',
+      'name' => 'Page',
+    ])->save();
+
+    FieldStorageConfig::create([
+      'field_name' => 'organisational_contact_info',
+      'entity_type' => 'node',
+      'type' => 'oe_translation_poetry_organisation_contact',
+    ])->save();
+
+    FieldConfig::create([
+      'field_name' => 'organisational_contact_info',
+      'entity_type' => 'node',
+      'bundle' => 'page',
+    ])->save();
+
+    FieldStorageConfig::create([
+      'field_name' => 'personal_contact_info',
+      'entity_type' => 'node',
+      'type' => 'oe_translation_poetry_personal_contact',
+    ])->save();
+
+    FieldConfig::create([
+      'field_name' => 'personal_contact_info',
+      'entity_type' => 'node',
+      'bundle' => 'page',
+    ])->save();
+  }
+
+  /**
+   * Tests the Organisational field type and formatter.
+   */
+  public function testOrganisationalContactItemField(): void {
+    /** @var \Drupal\node\NodeStorageInterface $node_storage */
+    $node_storage = $this->container->get('entity_type.manager')->getStorage('node');
+    $tests = [
+      'no-values' => [
+        'responsible' => '',
+        'author' => '',
+        'requester' => '',
+      ],
+      'only-author' => [
+        'responsible' => '',
+        'author' => 'Author',
+        'requester' => '',
+      ],
+      'only-requester' => [
+        'responsible' => '',
+        'author' => '',
+        'requester' => 'Requester',
+      ],
+      'only-responsible' => [
+        'responsible' => 'Responsible',
+        'author' => '',
+        'requester' => '',
+      ],
+      'all-values' => [
+        'responsible' => 'Responsible',
+        'author' => 'Author',
+        'requester' => 'Requester',
+      ],
+    ];
+
+    $node = $node_storage->create([
+      'type' => 'page',
+      'title' => 'Test page',
+    ]);
+    $node->save();
+
+    foreach ($tests as $case => $values) {
+      $node->set('organisational_contact_info', $values);
+      $node->save();
+      $node_storage->resetCache();
+      /** @var \Drupal\node\NodeInterface $node */
+      $node = $node_storage->load($node->id());
+      if ($case === 'no-values') {
+        $this->assertTrue($node->get('organisational_contact_info')->isEmpty());
+        continue;
+      }
+
+      $this->assertEquals($values, $node->get('organisational_contact_info')->first()->getValue());
+      $builder = $this->container->get('entity_type.manager')->getViewBuilder('node');
+      $build = $builder->viewField($node->get('organisational_contact_info'));
+      $output = $this->container->get('renderer')->renderRoot($build);
+      $this->assertContains(implode('<br />', $values), (string) $output);
+    }
+  }
+
+  /**
+   * Tests the Personal field type and formatter.
+   */
+  public function testPersonalContactItemField(): void {
```

Comment:
Both test methods do exactly the same thing, I believe we should be able to merge both an use a data provider method that provides the field name, the values and the expected empty() result.

Reply:
Yes, we can merge the tests, but no data provider.